### PR TITLE
Handle RPC error responses from RPC requests more quietly

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Don't log an ERROR level message when we receive an error response from an RPC request.

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2PeerManager.java [hunk 1]
```diff
@@ -98,7 +100,12 @@ public void onConnect(final Peer peer) {
           .sendStatus()
           .finish(
               () -> LOG.trace("Sent status to {}", peer.getId()),
-              (err) -> LOG.debug("Failed to send status to {}: {}", peer.getId(), err));
+              RootCauseExceptionHandler.builder()
+                  .addCatch(
+                      RpcException.class,
+                      err -> LOG.trace("Status message rejected by {}: {}", peer.getId(), err))
+                  .defaultCatch(
+                      err -> LOG.debug("Failed to send status to {}: {}", peer.getId(), err)));
```
Comment (ajsutton) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2PeerManager.java:108]: I got sick of the usual:
```
err -> {
  if (Throwables.getRootCause(err) instanceof RpcException) {
     // do one thing
  } else {
    // do something else
  });
```

I think this is slightly better but it's not as succinct as I was hoping for.

File: util/src/main/java/tech/pegasys/teku/util/async/RootCauseExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ */
+
+package tech.pegasys.teku.util.async;
+
+import com.google.common.base.Throwables;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class RootCauseExceptionHandler implements Consumer<Throwable> {
```
Comment (cemozerr) [inline util/src/main/java/tech/pegasys/teku/util/async/RootCauseExceptionHandler.java:21]: nice class.

**Review by cemozerr — APPROVED** [review]
LGTM.

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/Eth2OutgoingRequestHandler.java [hunk 3]
```diff
@@ -161,7 +163,7 @@ private void cancelRequest(
       return;
     }
 
-    LOG.debug("Cancel request: {}", error.getMessage());
+    LOG.trace("Cancel request: {}", error.getMessage());
```
Comment (mbaxter) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/Eth2OutgoingRequestHandler.java:166]: This seems useful at the debug level - no?  To figure out why requests aren't processing.

**Review by mbaxter — APPROVED** [review]

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/Eth2OutgoingRequestHandler.java (hunk 3 above)
Comment (ajsutton) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/Eth2OutgoingRequestHandler.java:166]: All the key cases already have a debug log before `cancelRequest` winds up getting called (and then the exception gets propagated back to the caller as well which usually winds up with more logging).

**Pull request merged**